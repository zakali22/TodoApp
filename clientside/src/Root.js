import React from "react";
import ReactDOM from "react-dom";
import "./index.scss";
import App from "./components/App";

import { createStore, applyMiddleware, compose } from "redux";
import { persistStore, persistReducer } from "redux-persist";
import storage from "redux-persist/lib/storage";
import thunk from "redux-thunk";
import rootReducer from "./reducers";
import { Provider } from "react-redux";
import { PersistGate } from "redux-persist/integration/react";

const initialState = {};
const middleware = [thunk];

const persistConfig = {
  key: "root",
  storage
};

const persistedReducer = persistReducer(persistConfig, rootReducer);

let store = createStore(
  persistedReducer,
  initialState,
  compose(
    applyMiddleware(...middleware),
    window.__REDUX_DEVTOOLS_EXTENSION__ && window.__REDUX_DEVTOOLS_EXTENSION__()
  )
);

let persistor = persistStore(store);

export default props => {
  return (
    <Provider store={store}>
      <PersistGate loading={null} persistor={persistor}>
        {props.children}
      </PersistGate>
    </Provider>
  );
};
