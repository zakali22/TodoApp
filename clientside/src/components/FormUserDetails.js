import React, { Component, Fragment } from "react";
import PropTypes from "prop-types";
import axios from "axios";

class FormUserDetails extends Component {
  state = {
    errors: []
  };
  continue = e => {
    e.preventDefault();
    const { first_name, last_name, email } = this.props.values;
    console.log(first_name, last_name, email);
    axios
      .post("/auth/checkPersonal", { first_name, last_name, email })
      .then(response => {
        if (response.data.success) {
          this.props.nextStep();
        } else {
          console.log(response.data);
          this.setState({
            errors: response.data.error
          });
        }
      });
  };

  onChange = e => {
    this.props.onChange(e);
  };

  render() {
    const { values } = this.props;
    const errorLists =
      this.state.errors.length > 0
        ? this.state.errors.map((error, i) => {
            return (
              <li key={i} className="error-list--item">
                {error.msg}
              </li>
            );
          })
        : [];
    return (
      <Fragment>
        <ul
          className="error-list"
          style={{ display: errorLists.length === 0 ? "none" : "block" }}
        >
          {errorLists}
        </ul>
        <form className="getstarted__local--form">
          <input
            value={values.first_name}
            placeholder="First name"
            name="first_name"
            onChange={this.onChange}
          />
          <input
            value={values.last_name}
            placeholder="Last name"
            name="last_name"
            onChange={this.onChange}
          />
          <input
            value={values.email}
            placeholder="Email"
            name="email"
            onChange={this.onChange}
          />
          <button onClick={this.continue}>Next</button>
        </form>
      </Fragment>
    );
  }
}

export default FormUserDetails;
