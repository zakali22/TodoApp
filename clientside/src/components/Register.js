import React from "react";
import PropTypes from "prop-types";
import { Link } from "react-router-dom";
import Google from "../Google.png";

const Register = props => {
  return (
    <div className="getstarted">
      <div className="container">
        <h1>Sign up</h1>
        <div className="container__register">
          <div className="getstarted__social">
            <a
              href="http://localhost:5000/auth/facebook"
              className="getstarted__social--facebook"
            >
              <button>
                <i className="fab fa-facebook" />
                <p>Continue with Facebook</p>
              </button>
            </a>
            <a
              href="http://localhost:5000/auth/google"
              className="getstarted__social--google"
            >
              <button>
                <img src={Google} />
                <p>Continue with Google</p>
              </button>
            </a>
          </div>
          <div className="getstarted__local">
            <h1>Local signing up </h1>
          </div>
        </div>
      </div>
    </div>
  );
};

export default Register;
