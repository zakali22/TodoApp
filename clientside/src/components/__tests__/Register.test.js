import React from "react";
import ReactDOM from "react-dom";
import { mount, shallow } from "enzyme";
import Register from "../Register";
import FormUserDetails from "../FormUserDetails";
import renderer from "react-test-renderer";
import { MemoryRouter } from "react-router-dom";
import Root from "../../Root";
import sinon from "sinon";

describe("Register", () => {
  test("has a valid snapshot", () => {
    const component = renderer.create(
      <MemoryRouter>
        <Register />
      </MemoryRouter>
    );
    const tree = component.toJSON();
    expect(tree).toMatchSnapshot();
  });
  describe("User types", () => {
    let wrapper, props, onChange;
    beforeEach(() => {
      wrapper = shallow(<Register />);
      props = {
        values: {
          first_name: "",
          last_name: "",
          email: ""
        },
        onChange: jest.fn()
      };
      onChange = sinon.stub(props, "onChange");
    });
    afterEach(() => {
      onChange.reset();
      wrapper.unmount();
    });
    it("sets state to values enter", () => {
      const childComponent = mount(<FormUserDetails {...props} />);
      const input = childComponent.find("input").first();
      input.simulate("change", {
        target: {
          value: "Something"
        }
      });
      childComponent.update();
      expect(input.props("value")).toEqual("Somthing");
    });
  });
});
