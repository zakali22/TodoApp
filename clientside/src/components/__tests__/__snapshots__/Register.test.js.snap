// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Register has a valid snapshot 1`] = `
<div
  className="getstarted"
>
  <div
    className="container"
  >
    <h1>
      Sign up
    </h1>
    <div
      className="container__register"
    >
      <div
        className="getstarted__social"
      >
        <a
          className="getstarted__social--facebook"
          href="http://localhost:5000/auth/facebook"
        >
          <button>
            <i
              className="fab fa-facebook"
            />
            <p>
              Continue with Facebook
            </p>
          </button>
        </a>
        <a
          className="getstarted__social--google"
          href="http://localhost:5000/auth/google"
        >
          <button>
            <img
              src="Google.png"
            />
            <p>
              Continue with Google
            </p>
          </button>
        </a>
      </div>
      <p
        id="divider"
      >
        or
      </p>
      <div
        className="getstarted__local"
      >
        <div
          className="container"
        >
          <ul
            className="error-list"
            style={
              Object {
                "display": "none",
              }
            }
          />
          <form
            className="getstarted__local--form"
          >
            <input
              name="first_name"
              onChange={[Function]}
              placeholder="First name"
              value=""
            />
            <input
              name="last_name"
              onChange={[Function]}
              placeholder="Last name"
              value=""
            />
            <input
              name="email"
              onChange={[Function]}
              placeholder="Email"
              value=""
            />
            <button
              onClick={[Function]}
            >
              Next
            </button>
          </form>
        </div>
      </div>
    </div>
  </div>
</div>
`;
